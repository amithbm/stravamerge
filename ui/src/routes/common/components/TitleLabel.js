import React from 'react';

const TitleLabel = ({text, subtitle, className}) => {
    const subtitleTag = subtitle ? (
        subtitle instanceof Date ? <span className="subtitle-label"> - {subtitle.toDateString()}</span> : <span className="subtitle-label"> - {subtitle}</span>) : undefined
    return <p className={className}>
        <span className="title-label">{text}</span>
        {subtitleTag}
    </p>
}

TitleLabel.propTypes = {
    text: React.PropTypes.string.isRequired,
    subtitle: React.PropTypes.string,
    className: React.PropTypes.string
}

export default TitleLabel
