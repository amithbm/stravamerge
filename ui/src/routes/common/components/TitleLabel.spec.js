import React from 'react'
import {shallow} from 'enzyme'
import TitleLabel from './TitleLabel'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<TitleLabel />', () => {
        it('renders as expected', () => {
            const component =  shallow(<TitleLabel text="Hello World" subtitle="Too" className="aclass" />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})

