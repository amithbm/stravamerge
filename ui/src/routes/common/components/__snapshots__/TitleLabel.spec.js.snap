exports[`components <TitleLabel /> renders as expected 1`] = `
<p
  className="aclass">
  <span
    className="title-label">
    Hello World
  </span>
  <span
    className="subtitle-label">
     - 
    Too
  </span>
</p>
`;
